File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        vector<int> lmx,rmx;
        int n = height.size();
        int lx = height[0];
        int rx = height[n-1];
        int ans =0;
        for(int i =0 ;i<n;i++){
            lx = max(height[i],lx);
            lmx.push_back(lx);
        }
        for(int i=n-1;i>=0;i-- ){
            rx =  max(height[i],rx);
            rmx.push_back(rx);
        }
        reverse(rmx.begin(),rmx.end());
        for(int i =0;i<n;i++){
            std::cout<<lmx[i]<<"--"<<rmx[i]<<endl;
            ans += min(lmx[i],rmx[i])-height[i];
        }
        return ans;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var maxArea = 0
        var i = 0
        var j = height.size-1
        while(i<j){
            maxArea = max(((j-i)*(min(height[i],height[j]))),maxArea)
            if(height[i]<height[j]){
                i++
            }else{
                j--
            }
        }
        return maxArea
    }
}
